### PR TITLE
Revert Revert "[Placement Group] Support infeasible placement groups for Placement Group."

### PR DESCRIPTION
**Pull request by clay4megtr**
## Why are these changes needed?

revert #16509

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by clay4444** [comment]
I will check all the failed ut this time.

File: src/ray/gcs/gcs_server/gcs_actor_distribution.cc [hunk 1]
```diff
@@ -92,7 +92,8 @@ GcsBasedActorScheduler::AllocateNewActorWorkerAssignment(
 
 NodeID GcsBasedActorScheduler::AllocateResources(const ResourceSet &required_resources) {
   auto selected_nodes =
-      gcs_resource_scheduler_->Schedule({required_resources}, SchedulingType::SPREAD);
+      gcs_resource_scheduler_->Schedule({required_resources}, SchedulingType::SPREAD)
```
Comment (clay4444) [inline src/ray/gcs/gcs_server/gcs_actor_distribution.cc:95]: The first conflict when trying to revert this pr.

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc [hunk 2]
```diff
@@ -296,8 +302,8 @@ void GcsPlacementGroupManager::SchedulePendingPlacementGroups() {
       MarkSchedulingStarted(placement_group_id);
```
Comment (clay4444) [inline src/ray/gcs/gcs_server/gcs_placement_group_manager.cc:302]: the second is here about the `is_new_placement_group_scheduled` flag.

**Comment by wuisawesome** [comment]
tests passing, lgtm

**Review by wuisawesome — APPROVED** [review]

**Pull request merged**